### PR TITLE
use react-test-library for testing components

### PR DESCRIPTION
**Pull request by piesieomane**
In this Pull request the following list below was implemented;
- Wrote unit tests for the files `operate.js` and `calculate.js` using Jest.
- Wrote test for all components using `react-test-renderer`

### CODE REVIEW

File: src/logic/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import operate from './operate';
+
+test('should take two numbers and an operation and retrun the result', () => {
+  expect(operate(1, 2, '+')).toBe('3');
+  expect(operate(1, 2, '-')).toBe('-1');
+  expect(operate(1, 2, 'x')).toBe('2');
+  expect(operate(1, 2, '÷')).toBe('0.5');
+  expect(operate(1, 2, '%')).toBe('1');
+});
```
Comment (Nemwel-Boniface) [inline src/logic/operate.test.js]: You have both done amazing job writing tests for your operate.js file. However, kindly note that the purpose of writing your tests is to simulate all possible scenarios that would arise and try to write a test for them to handle such scenarios. This said it would be a good thing to add tests for the following scenarios:

- [ ] Add tests for scenarios where a user performs a division by a zero
- [ ] Add tests for scenarios where the user uses the modulus operator with zero
- [ ] Add tests for scenarios where the user makes operations with negative numbers

File: src/components/components.test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  expect(tree).toMatchSnapshot();
+});
+
+it('renders correctly', () => {
+  <Router>
+    const tree = renderer.create(
+    <Nav />
+    ).toJSON(); expect(tree).toMatchSnapshot();
+  </Router>;
+});
```
Comment (Nemwel-Boniface) [inline src/components/components.test.js]: You have both done well in writing your tests for your project components. Currently, I can see that you are both not testing user interaction eg when a user clicks on a specific link to take them to a specific page, does it render the expected results? You can have a look at [this resource](https://www.freecodecamp.org/news/8-simple-steps-to-start-testing-react-apps-using-react-testing-library-and-jest/) which will help you write better tests for your project and get to learn in the process.

- [ ] Kindly break the tests per component and eg have all the tests for a specific component in its own .test.js file. eg all tests for Home.js should be in the Home.test.js file
- [ ] Kindly make use of React Testing Library to simulate user interaction
- [ ] Kindly make sure to write tests for all the components in individual .test.js files.

File: src/logic/calculate.test.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import calculate from './calculate';
+
+describe('calculate', () => {
+  it('should return an object', () => {
+    const result = calculate({ total: 0, next: null, operation: null }, 'AC');
+    expect(result).toEqual({ total: 0, next: null, operation: null });
+  });
+});
```
Comment (Nemwel-Boniface) [inline src/logic/calculate.test.js]: - [ ] Kindly add more tests for the Calculate.js file. There is a lot that you could test for in this file eg of a test you can perform included below:

```
test('test " - " button', () => {
  const obj = {
    total: '100',
    next: '4',
    operation: '-',
  };
  const output = calculate(obj, '=');
  expect(output).toEqual({ total: '96', next: null, operation: null });
});
```

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @piesieomane  and @digia3d            :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

You have both done an amazing job adding tests for your project :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

- [ ] KIndly add more tests for your project testing user interaction with your application making use of the React Testing Library.

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hello @piesieomane  👋,

_This is an additional review as a result of the second opinion request._

_Great job so far 👏 💯 👍. However, there are few issues we need to address before we move on. Please, check the comments below._

## Required Changes ♻️
 - [ ] As request by the previous reviewer, the project requires you to Use `React Testing Library` to simulate user interaction and to test the actual content of the components. something like what happens if I click on this button? or what text in the calculator display?.. but, currently, you only tested the snapshots which is not enough. Kindly refer to previous review and implement the requested changes.
 
Screen Shot 2022-08-10 at 22 10 38


Keep rocking 🚀!

File: src/components/__snapshots__/Calculator.test.js.snap [hunk 4]
```diff
@@ -0,0 +1,142 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (BenMukebo) [inline src/components/__snapshots__/Calculator.test.js.snap:1]: - [OPTIONAL] According to the project requirements, There should be the usage of Jest snapshots for testing all the components. shower you only have Calculator, Home, and Quote snapshots testing. Please you can consider also adding the Jest snapshots for testing the `Nav` component.

File: src/components/Nav.jsx [hunk 5]
```diff
@@ -1,24 +1,28 @@
 /* eslint-disable */
 import React from "react";
-import { Link } from "react-router-dom";
+// import { Link } from "react-router-dom";
```
Comment (BenMukebo) [inline src/components/Nav.jsx:3]: - [OPTIONAL] Please do not leave unused code as a comment in your codebase. this will make your code much noisy, It is always a good practice to keep your code clean from such comments. 😉 👍🏻

**Review by BenMukebo — APPROVED** [review]
### Hi @piesieomane, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 




## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [OPTIONAL] Kindly, Add a descriptive README file to your project, it will be great to add instructions on how to run the tests. As this will be of help to non-technical users. This includes `npm run test` etc

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
